src/main/java/remer/movieexplorer/MovieController.java

```diff
@@ -0,0 +1,125 @@
+
+        if (response.search == null || response.search.isEmpty())
+        {
+            showError("No results found.");
+            return;
+        }
+
+        int count = 0;
+        for (Movie movie : response.search)
+        {
+            if (count >= 9)
+            {
+                break; // Show only 9 posters
+            }
+
+            JLabel label = new JLabel("<html>" + movie.title + " (" + movie.year + ")</html>", JLabel.CENTER);
+            label.setVerticalTextPosition(JLabel.BOTTOM);
+            label.setHorizontalTextPosition(JLabel.CENTER);
+
+            // Set the poster image if available
+            if (movie.poster != null && !movie.poster.equalsIgnoreCase("N/A"))
+            {
+                try
+                {
+                    Image img = ImageIO.read(new URL(movie.poster));
+                    if (img != null)
+                    {
+                        ImageIcon icon = new ImageIcon(img.getScaledInstance(150, 220, Image.SCALE_SMOOTH));
+                        label.setIcon(icon);
+                    }
+                } catch (IOException e)
+                {
+                    System.err.println("Could not load poster for " + movie.title + ": " + e.getMessage());
+                }
+            }
+
+            // Add mouse listener for detail popup
+            label.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
+            label.addMouseListener(new java.awt.event.MouseAdapter()
+            {
+                @Override
+                public void mouseClicked(java.awt.event.MouseEvent evt)
+                {
+                    showMovieDetail(movie.imdbId);
+                }
+            });
+
+            gridPanel.add(label);
+            count++;
+        }
+        // Fill empty cells if less than 9 results
+        while (count < 9)
+        {
+            gridPanel.add(new JLabel());
+            count++;
+        }
+        gridPanel.revalidate();
+        gridPanel.repaint();
+    }
+
+    private void showMovieDetail(String imdbId)
+    {
+        service.getMovieDetail(imdbId, apiKey.get())
+                .subscribeOn(Schedulers.io())
+                .observeOn(Schedulers.from(SwingUtilities::invokeLater))
+                .subscribe(
+                        movieDetail -> {
+                            MovieDetailFrame detailFrame = new MovieDetailFrame(movieDetail);
+                            detailFrame.setLocationRelativeTo(null);
+                            detailFrame.setVisible(true);
+                        },
+                        e -> JOptionPane.showMessageDialog(gridPanel, "Error loading details: " + e.getMessage())
+                );
+    }
+
+    private void showError(String message)
+    {
+        gridPanel.removeAll();
+        gridPanel.setLayout(new GridLayout(1, 1));
+        gridPanel.add(new JLabel(message, JLabel.CENTER));
+        gridPanel.revalidate();
+        gridPanel.repaint();
```

Comment:
Don't modify the output.
Use JOptionPane.showMessageDialog(... , JOptionPane.ERROR_MESSAGE)

https://www.tutorialspoint.com/swingexamples/show_error_message_dialog.htm